indexer/ai/src/main/java/com/google/cloudsearch/LocalRepository.java

```diff
@@ -0,0 +1,183 @@
+        //Read and store file paths from the resource folder
+
+        ConfigValue<String> folderName = Configuration.getValue("resources.names",null,Configuration.STRING_PARSER);
+        log.info(folderName.get());
+        File[] files = getAllFiles(folderName.get());
+        for (File f : files) {
+            log.info(f.getPath());
+            allFileNames.add(f.getPath());
+        }
+        //Parse the JSON
+        ConfigValue<String> AISkillConfig = Configuration.getValue("enrichment.config", null, Configuration.STRING_PARSER);
+        JSONParser parser = new JSONParser();
+        try {
+            Object aiConfig = parser.parse(new FileReader(String.valueOf(AISkillConfig.get())));
+            this.skillSet = new AISkillSet((JSONObject) aiConfig);
+        } catch (IOException e) {
+            e.printStackTrace();
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public CheckpointCloseableIterable<ApiOperation> getIds(@Nullable byte[] bytes) throws RepositoryException {
+        return null;
+    }
+
+    @Override
+    public CheckpointCloseableIterable<ApiOperation> getChanges(@Nullable byte[] bytes) throws RepositoryException {
+        return null;
+    }
+
+    @Override
+    public CheckpointCloseableIterable<ApiOperation> getAllDocs(@Nullable byte[] bytes) throws RepositoryException {
+        //Iterate through all resources and execute skills for each resource
+        /*
+            Pseudo Code:
+
+            for each resource:
+                for each skill in aiSkillSet:
+                    executeSkill(resourcePath)
+         */
+
+        Iterator<ApiOperation> allDocs = IntStream.range(0, allFileNames.size())
+                .mapToObj(id -> {
+                    try {
+                        return buildDocuments(id);
+                    } catch (Exception e) {
+                        e.printStackTrace();
+                    }
+                    return null;
+                })
+                .iterator();
+
+        CheckpointCloseableIterable<ApiOperation> iterator =
+                new CheckpointCloseableIterableImpl.Builder<>(allDocs).build();
+
+        return iterator;
+
+    }
+    private ApiOperation buildDocuments(int id){
+        log.info("inside");
+        Acl acl = new Acl.Builder()
+                .setReaders(Collections.singletonList(Acl.getCustomerPrincipal()))
+                .build();
+        String filepath = this.allFileNames.get(0);
+
+        List<AISkill> skillList = (List<AISkill>) skillSet.getSkillSet();
+        Multimap<String, Object> structuredData = ArrayListMultimap.create();
+        Iterator<AISkill> skillIterator = skillList.iterator();
+        while(skillIterator.hasNext()){
+            AISkill skill = skillIterator.next();
+            log.info(skill.getAISkillName());
+            structuredData = skill.executeSkill(filepath, structuredData);
```

Comment:
I just thought if the file is too big, it will be inefficient to pass such a big string down the function call chain, that's why I used the path instead.

Though now that I have evolved the code to make it more like a library, the user may want to pass part of the file for the AI skills instead of the entire file. But one more thing is that for custom skill, if the content is too big we may want to pass the bucket URI instead of the actual content, that's why I also wanted to keep the file path.

May be we can make it content for local repository or bucket URI if the files are in cloud storage?